injected/src/detectors/detector-init.js

```diff
@@ -0,0 +1,133 @@
+/**
+ * Detector Initialization
+ *
+ * Reads bundledConfig and registers detectors with the detector service.
+ * Called during content-scope-features load phase.
+ */
+
+import { registerDetector } from './detector-service.js';
+import { DEFAULT_DETECTOR_SETTINGS } from './default-config.js';
+import { createBotDetector } from './detections/bot-detection.js';
+import { createFraudDetector } from './detections/fraud-detection.js';
+import { matchesDomainPatterns } from './utils/detection-utils.js';
+
+/**
+ * Check if gates should run for this call
+ * @param {Object} options - Options passed to getData
+ * @param {boolean} [options._autoRun] - Internal flag indicating this is an auto-run call
+ * @param {string[]} [domains] - Domain patterns to check
+ * @param {Function} [shouldRun] - Custom gate function
+ * @returns {boolean} True if gates pass, false otherwise
+ */
+function checkGates(options, domains, shouldRun) {
+    // Only check gates for auto-run calls
+    // Manual calls bypass gates by default
+    if (!options?._autoRun) {
+        return true;
+    }
+
+    // Check domain gate
+    if (!matchesDomainPatterns(domains)) {
+        return false;
+    }
+
+    // Check custom gate function if provided
+    if (shouldRun && !shouldRun()) {
+        return false;
+    }
+
+    return true;
+}
+
+/**
+ * Create a gated detector registration that checks domain and custom gates
+ * Gates only apply for auto-run calls (when _autoRun: true is passed)
+ * Manual calls bypass gates by default
+ *
+ * @param {Object} registration - The detector registration object
+ * @param {Function} registration.getData - Function to get detector data
+ * @param {Function} [registration.shouldRun] - Optional gate function
+ * @param {Function} [registration.refresh] - Optional refresh function
+ * @param {Function} [registration.teardown] - Optional teardown function
+ * @param {string[]} [domains] - Optional array of domain patterns
+ * @returns {Object} Gated detector registration
+ */
+function createGatedDetector(registration, domains) {
+    const { getData, shouldRun, refresh, teardown } = registration;
+
+    return {
+        getData: async (options) => {
+            // Check gates (only for auto-run, manual calls bypass)
+            if (!checkGates(options, domains, shouldRun)) {
+                return null;
+            }
+
+            // All gates passed, run the detector
+            return await getData(options);
+        },
+        refresh: refresh ? async (options) => {
+            if (!checkGates(options, domains, shouldRun)) {
+                return null;
+            }
+            return await refresh(options);
```

Comment:
I don't think bot-detection or fraud-detection has refresh but this I believe exposes it as having one meaning we will preference refresh right?

In:
```
    const runner = registration.refresh ?? registration.getData;
```
We're preferencing this function which is a noop as I understand it.